tutorials/training_ppi.ipynb

```diff
@@ -0,0 +1,6991 @@
+    "        node0_features = node_features[node0_indices]\n",
+    "        node1_features = node_features[node1_indices]\n",
+    "        message_input = torch.cat([node0_features, node1_features, edge_features], dim=1)\n",
+    "        messages_per_neighbour = self._edge_mlp(message_input)\n",
+    "        # aggregate messages\n",
+    "        out = torch.zeros(node_features.shape[0], messages_per_neighbour.shape[1]).to(node_features.device)\n",
+    "        message_sums_per_node = scatter_sum(messages_per_neighbour, node0_indices, dim=0, out=out)\n",
+    "        # update nodes\n",
+    "        node_input = torch.cat([node_features, message_sums_per_node], dim=1)\n",
+    "        node_output = self._node_mlp(node_input)\n",
+    "        return node_output\n",
+    "\n",
+    "    class NaiveNetwork(Module):\n",
+    "\n",
+    "        def __init__(self, input_shape: int, output_shape: int, input_shape_edge: int):\n",
+    "            \"\"\"\n",
+    "            Args:\n",
+    "                input_shape (int): Number of node input features.\n",
+    "                output_shape (int): Number of output value per graph.\n",
+    "                input_shape_edge (int): Number of edge input features.\n",
+    "            \"\"\"\n",
+    "            super().__init__()\n",
+    "            self._external1 = NaiveConvolutionalLayer(input_shape, input_shape_edge)\n",
+    "            self._external2 = NaiveConvolutionalLayer(input_shape, input_shape_edge)\n",
+    "            hidden_size = 128\n",
+    "            self._graph_mlp = Sequential(Linear(input_shape, hidden_size), ReLU(), Linear(hidden_size, output_shape))\n",
+    "\n",
+    "        def forward(self, data):\n",
+    "            external_updated1_node_features = self._external1(data.x, data.edge_index, data.edge_attr)\n",
+    "            external_updated2_node_features = self._external2(external_updated1_node_features, data.edge_index, data.edge_attr)\n",
+    "            means_per_graph_external = scatter_mean(external_updated2_node_features, data.batch, dim=0)\n",
+    "            graph_input = means_per_graph_external\n",
+    "            z = self._graph_mlp(graph_input)\n",
+    "            return z\n",
+    "  ```\n",
+    "- `class_weights` is used in classification tasks only, to assign class weights based on the training dataset content. In this case we have a balanced dataset (50% 0 and 50% 1), so it doesn't make sense to use it. It defaults to False. \n",
+    "- `cuda` and `ngpu` are used for indicating whether to use CUDA and how many GPUs. By default, CUDA is not used and `ngpu` is 0.\n",
+    "- The user can specify a deeprankcore exporter or a custom one in `output_exporters` parameter, together with the path where to save the results. Exporters are used for storing predictions information collected later on during training and testing. We will see later how to read the results saved by `HDF5OutputExporter`."
```

Comment:
There's a shift in language here from "let's do this and that" to "the user can"